sopel/plugins/capabilities.py

```diff
@@ -0,0 +1,477 @@
+
+        Requests made are stored as requested; others are ignored::
+
+            >>> # manager.register => cap1 and cap2, not cap3
+            >>> manager.request_available(bot, ('cap1', 'cap3'))
+            >>> manager.is_requested(('cap1',))
+            True
+            >>> manager.is_requested(('cap2',))
+            False
+            >>> manager.is_requested(('cap3',))
+            False
+
+        .. important::
+
+            The capability request ``('cap1', '-cap2')`` means "enable cap1 and
+            disable cap2", and the request will be acknowledged or denied at
+            once. If the server doesn't advertise any of these capabilities,
+            the client **should not** send a request.
+
+            As a result, Sopel will send a request to enable or disable
+            a capability only if it is advertised first. If a request is
+            never made, its callback will never be called, because there
+            won't be any related ACK/NAK message from the server.
+
+            Plugin authors should not use the request's callback to activate or
+            deactivate features, and instead check the bot's capabilities every
+            time they need to.
+
+            The only exception to that is when the plugin needs to perform an
+            operation while negotiating the capabilities (such as SASL auth).
+
+        """
+        # Requesting capabilities when they are available
+        capabilities = set(available_capabilities)
+        for cap_req, plugin_requests in self._registered.items():
+            # request only if all are available
+            cap_req_enables = {
+                cap.lstrip('-')
+                for cap in cap_req
+            }
+            text = ' '.join(cap_req)
+            if cap_req_enables <= capabilities:
+                self._registered[cap_req] = {
+                    plugin_name: (handler_info[0], False)
+                    for plugin_name, handler_info in plugin_requests.items()
+                }
+                self._requested.add(cap_req)
+                LOGGER.debug('Capability negotiation request: "%s".', text)
+                bot.write(('CAP', 'REQ'), text=text)
+            else:
+                LOGGER.debug(
+                    'Unable to negotiate capability request: "%s".', text,
+                )
+
+    def resume(
+        self,
+        request: Iterable[str],
+        plugin_name: str,
+    ) -> Tuple[bool, bool]:
+        """Resume the registered plugin capability request.
+
+        :return: a 2-value tuple with (was completed, is completed)
+
+        The capability request, for that plugin, will be marked as done,
+        and the result will be about the capability negotiation process:
+
+        * was it already completed before the resume?
+        * is it completed now?
+
+        If the capability request cannot be found for that plugin, the result
+        value remains the same (it stay uncomplete or complete)
```

Comment:
```suggestion
        value remains the same (it stays incomplete or complete)
```

Rats. Missed this one.